Content.Server/Damage/Systems/GodmodeSystem.cs

```diff
@@ -1,98 +1,35 @@
 using Content.Server.Atmos.Components;
-using Content.Server.Damage.Components;
-using Content.Shared.Damage;
+using Content.Shared.Damage.Components;
 using Content.Shared.Damage.Systems;
-using Content.Shared.FixedPoint;
-using Content.Shared.Rejuvenate;
-using Content.Shared.StatusEffect;
-using JetBrains.Annotations;
 
-namespace Content.Server.Damage.Systems
+namespace Content.Server.Damage.Systems;
+
+public sealed class GodmodeSystem : SharedGodmodeSystem
 {
-    [UsedImplicitly]
-    public sealed class GodmodeSystem : EntitySystem
+    public override void EnableGodmode(EntityUid uid, GodmodeComponent? godmode = null)
     {
-        [Dependency] private readonly DamageableSystem _damageable = default!;
-
-        public override void Initialize()
-        {
-            base.Initialize();
-
-            SubscribeLocalEvent<GodmodeComponent, BeforeDamageChangedEvent>(OnBeforeDamageChanged);
-            SubscribeLocalEvent<GodmodeComponent, BeforeStatusEffectAddedEvent>(OnBeforeStatusEffect);
-            SubscribeLocalEvent<GodmodeComponent, BeforeStaminaDamageEvent>(OnBeforeStaminaDamage);
-        }
-
-        private void OnBeforeDamageChanged(EntityUid uid, GodmodeComponent component, ref BeforeDamageChangedEvent args)
-        {
-            args.Cancelled = true;
-        }
-
-        private void OnBeforeStatusEffect(EntityUid uid, GodmodeComponent component, ref BeforeStatusEffectAddedEvent args)
-        {
-            args.Cancelled = true;
-        }
+        base.EnableGodmode(uid, godmode);
```

Comment:
this also needs to actually check if godmode was already enabled and then fetch the component after enabling